python/RceG3/_RceEthernet.py

```diff
@@ -0,0 +1,291 @@
+            linkedSet = lambda value: self.MacAddressRaw.set(int(''.join(reversed(value.split(':'))), 16))
+        
+        self.add(pr.RemoteVariable(
+            name = 'IpAddressRaw',
+            offset = 0x01C,
+            bitOffset = 0,
+            bitSize = 32,
+            base = pr.UInt,
+            mode = 'RW'))
+
+        self.add(pr.LinkVariable(
+            name = 'IpAddress',
+            mode = 'RW',
+            dependencies = [self.IpAddressRaw]
+            linkedGet = lambda: '.'.join(f'{b:d}' for b in self.IpAddressRaw.value().to_bytes(4, 'little'))
+            linkedSet = lambda value: self.IpAddressRaw.set(int.from_bytes((int(x) for x in value.split('.')), 'little')            
+        
+        self.add(pr.RemoteVariable(
+            name = 'PhyStatus',
+            offset = 0x020,
+            bitOffset = 0,
+            bitSize = 8,
+            base = pr.UInt,
+            mode = 'RO'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'PhyDebug',
+            offset = 0x024,
+            bitOffset = 0,
+            bitSize = 6,
+            base = pr.UInt,
+            mode = 'RO'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'TxShift'
+            offset = 0x038,
+            bitOffset = 0,
+            bitSize = 4,
+            base = pr.UInt,
+            mode = 'RW'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'RxShift',
+            offset = 0x038,
+            bitOffset = 4,
+            bitSize = 4,
+            base = pr.UInt,
+            mode = 'RW'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'FiltEnable',
+            offset = 0x038,
+            bitOffset = 16,
+            bitSize = 1,
+            base = pr.Bool,
+            mode = 'RW'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'IpChecksumEnable',
+            offset = 0x038,
+            bitOffset = 17,
+            bitSize = 1,
+            base = pr.Bool,
+            mode = 'RW'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'TcpChecksumEnable',
+            offset = 0x038,
+            bitOffset = 18,
+            bitSize = 1,
+            base = pr.Bool
+            mode = 'RW'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'UdpChecksumEnable',
+            offset = 0x038,
+            bitOffset = 19,
+            bitSize = 1,
+            base = pr.Bool,
+            mode = 'RW'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'DropOnPause',
+            offset = 0x038,
+            bitOffset = 20,
+            bitSize = 1,
+            base = pr.Bool,
+            mode = 'RW'))
+        
+        self.add(pr.RemoteVariable(
+            name = 'EthHeaderSize',
+            offset = 0x03C,
+            bitOffset = 0,
+            bitSize = 16,
+            base = pr.UInt,
+            mode = 'RW'))
```

Comment:
All the registers in this module (expect for the IP address) should be 'RO' because they are set by the kernel and could potential brick the ETH connnection